src/control/server/ctl_storage.go

```diff
@@ -58,32 +58,40 @@ func NewMockStorageControlService(log logging.Logger, engineCfgs []*engine.Confi
 	)
 }
 
-// Setup delegates to Storage implementation's Setup methods.
-func (c *StorageControlService) Setup() error {
+// Setup performs storage discovery and validates existence of configured devices.
+func (c *StorageControlService) Setup() {
 	if _, err := c.ScmScan(storage.ScmScanRequest{}); err != nil {
 		c.log.Debugf("%s\n", errors.Wrap(err, "Warning, SCM Scan"))
 	}
 
-	// don't scan if using emulated NVMe
-	// TODO: this should probably be verified and scan per engine
+	var isNotNVME bool
+	var cfgBdevs []string
+
 	for _, storageCfg := range c.instanceStorage {
 		for _, tierCfg := range storageCfg.Tiers.BdevConfigs() {
 			if tierCfg.Class != storage.ClassNvme {
-				return nil
+				isNotNVME = true
+				continue
```

Comment:
Hm, so is the intent to abort if any bdev config is emulated? Or only if all of them are?

If we want to abort if any is emulated, you could return here instead of using the variable.

Reply:
done